### PR TITLE
Change HTML structure of image selection components according to HTML standards

### LINKED ISSUE #8811: Change HTML Structure of Image Selection components with Kacper's proposals.
The HTML of the image attribute type is not ideal, we should make use of what HTML offers us.

Use a button element instead of a div – a div with the tabindex/click/keyup/focus property is bad practice.

Ask Kacper for review.

Execute Cypress locally and if necessary adapt selectors

### PR DESCRIPTION
**Pull request by Larisa-Staroverova**

### CODE REVIEW

File: feature-libs/product/configurators/common/src/components/config-attribute/config-attribute-types/config-attribute-multi-selection-image/config-attribute-multi-selection-image.component.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
-<div id="{{ createAttributeIdForConfigurator(attribute) }}" class="row">
-  <div
+<ul id="{{ createAttributeIdForConfigurator(attribute) }}" class="row">
+  <li
     *ngFor="let value of attribute.values; let i = index"
     id="{{ createAttributeValueIdForConfigurator(attribute, value.valueCode) }}"
     (click)="onSelect(i)"
```
Comment (kacperknapik) [inline feature-libs/product/configurators/common/src/components/config-attribute/config-attribute-types/config-attribute-multi-selection-image/config-attribute-multi-selection-image.component.html]: only natively interactable elements (buttons, link, etc) should be clickable. please remove `(click)` event from here, and use a button or something similar inside this `li` element

Comment (kacperknapik) [reply feature-libs/product/configurators/common/src/components/config-attribute/config-attribute-types/config-attribute-multi-selection-image/config-attribute-multi-selection-image.component.html]: also `(keyup)`, `role=checkbox`, `cxFocus`, `aria-checked`, and `tabindex` shouldn't be here

File: feature-libs/product/configurators/common/src/components/config-attribute/config-attribute-types/config-attribute-single-selection-image/config-attribute-single-selection-image.component.html [hunk 2]
```diff
@@ -1,5 +1,5 @@
-<div id="{{ createAttributeIdForConfigurator(attribute) }}" class="row">
-  <div
+<ul id="{{ createAttributeIdForConfigurator(attribute) }}" class="row">
+  <li
```
Comment (kacperknapik) [inline feature-libs/product/configurators/common/src/components/config-attribute/config-attribute-types/config-attribute-single-selection-image/config-attribute-single-selection-image.component.html]: same as above

**Review by kacperknapik — CHANGES_REQUESTED** [review]

File: feature-libs/product/configurators/common/src/components/config-attribute/config-attribute-types/config-attribute-multi-selection-image/config-attribute-multi-selection-image.component.html (hunk 1 above)
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/src/components/config-attribute/config-attribute-types/config-attribute-multi-selection-image/config-attribute-multi-selection-image.component.html]: Done!

File: feature-libs/product/configurators/common/components/attribute/types/radio-button/configurator-attribute-radio-button.component.ts [hunk 3]
```diff
@@ -19,80 +21,36 @@ export class ConfiguratorAttributeRadioButtonComponent
...
-      this.attributeRadioButtonForm.value !== null &&
-      this.changeTriggeredByKeyboard === true
-    ) {
-      this.submitValue(this.attributeRadioButtonForm.value);
-    }
-  }
-
-  onKeyUp() {
-    this.changeTriggeredByKeyboard = true;
+  constructor(protected configUtils: ConfiguratorStorefrontUtilsService) {
```
Comment (schuppell) [inline feature-libs/product/configurators/common/components/attribute/types/radio-button/configurator-attribute-radio-button.component.ts]: configUtils is not used

File: feature-libs/product/configurators/common/components/attribute/types/single-selection-image/configurator-attribute-single-selection-image.component.ts [hunk 4]
```diff
@@ -22,37 +23,29 @@ export class ConfiguratorAttributeSingleSelectionImageComponent
...
 
   @Input() attribute: Configurator.Attribute;
-  @Input() group: string;
   @Input() ownerKey: string;
 
   @Output() selectionChange = new EventEmitter<ConfigFormUpdateEvent>();
 
-  ngOnInit() {
-    this.attributeRadioButtonForm.setValue(this.attribute.selectedSingleValue);
+  constructor(protected configUtils: ConfiguratorStorefrontUtilsService) {
```
Comment (schuppell) [inline feature-libs/product/configurators/common/components/attribute/types/single-selection-image/configurator-attribute-single-selection-image.component.ts]: configUtils is not used

File: feature-libs/product/configurators/common/components/service/configurator-storefront-utils.service.ts [hunk 5]
```diff
@@ -105,4 +105,35 @@ export class ConfiguratorStorefrontUtilsService {
...
+    return id?.substring(0, index);
+  }
+
+  /**
+   * Returns true if the focus changed from one attribute to any other element.
+   * Returns false if the focus change occurred inside the same attribute (i.e. from one value to another of the same attribute)
+   *
+   * @param {FocusEvent} event FocusEvent to be checked
+   */
+  attributeLostFocus(event: FocusEvent): boolean {
```
Comment (schuppell) [inline feature-libs/product/configurators/common/components/service/configurator-storefront-utils.service.ts]: Is this method used at all?

File: feature-libs/product/configurators/common/styles/_configurator-attribute-multi-selection-image.scss [hunk 6]
```diff
@@ -1,39 +1,3 @@
...
-      -webkit-appearance: none;
-      visibility: hidden;
-      left: 25px;
-    }
-
-    .cx-configurator-attribute-value-label {
-      text-align: center;
-    }
-  }
+  @include cx-configurator-attribute-selection-image();
```
Comment (schuppell) [inline feature-libs/product/configurators/common/styles/_configurator-attribute-multi-selection-image.scss:2]: Nice!

File: feature-libs/product/configurators/common/styles/_configurator-attribute-single-selection-image.scss [hunk 7]
```diff
@@ -1,39 +1,3 @@
...
-      -webkit-appearance: none;
-      visibility: hidden;
-      left: 25px;
-    }
-
-    .cx-configurator-attribute-value-label {
-      text-align: center;
-    }
-  }
+  @include cx-configurator-attribute-selection-image();
```
Comment (schuppell) [inline feature-libs/product/configurators/common/styles/_configurator-attribute-single-selection-image.scss:2]: Nice!

**Review by schuppell — CHANGES_REQUESTED** [review]

File: feature-libs/product/configurators/common/components/attribute/types/single-selection-image/configurator-attribute-single-selection-image.component.ts (hunk 4 above)
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/components/attribute/types/single-selection-image/configurator-attribute-single-selection-image.component.ts]: Done!

File: feature-libs/product/configurators/common/components/service/configurator-storefront-utils.service.ts (hunk 5 above)
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/components/service/configurator-storefront-utils.service.ts]: Done! The method is removed with the corresponding tests

File: feature-libs/product/configurators/common/styles/_configurator-attribute-multi-selection-image.scss (hunk 6 above)
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/styles/_configurator-attribute-multi-selection-image.scss:2]: Thanks!

File: feature-libs/product/configurators/common/styles/_configurator-attribute-single-selection-image.scss (hunk 7 above)
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/styles/_configurator-attribute-single-selection-image.scss:2]: Thanks!

**Pull request merged**